src/status_im/chat/models.cljs

```diff
@@ -282,14 +288,19 @@
 
 (fx/defn start-public-chat
   "Starts a new public chat"
-  [cofx topic {:keys [dont-navigate?] :as opts}]
-  (fx/merge cofx
-            (add-public-chat topic)
-            #(when-not dont-navigate?
-               (navigate-to-chat % topic opts))
-            (public-chat/join-public-chat topic)
-            (when platform/desktop?
-              (desktop.events/change-tab :home))))
+  [cofx topic {:keys [dont-navigate? sync?] :or {sync? true} :as opts}]
+  (if (active-chat? cofx topic)
+    (when-not dont-navigate?
+      (navigate-to-chat cofx topic opts))
+    (fx/merge cofx
+              (add-public-chat topic)
+              #(when-not dont-navigate?
+                 (navigate-to-chat % topic opts))
+              (public-chat/join-public-chat topic)
+              #(when platform/desktop?
+                 (desktop.events/change-tab % :home))
+              #(when sync?
+                 {:dispatch [:chat.ui/sync-public-chat topic]}))))
```

Comment:
this shouldn't dispatch an event but simply call `(pairing/sync-public-chat topic))` directly

Reply:
as a rule of thumb if your event name sounds like an action, it probably doesn't need an event. Events are about what happened not what will happen

Reply:
It causes a circular dependency though, that's the only reason I did it like that

Reply:
circular dependencies means that some namespaces might be doing more that what they should but to avoid having to fix that in that PR, I would say it is ok to have a fx/merge in events.cljs since it was already the case.

Reply:
but then it would not fix https://github.com/status-im/status-react/issues/7992 case

Reply:
It has been a while since I worked on this and I have a bit of a broken/slow build system hindering work at the moment, so off the top of my head I'm now wondering if moving `sync-public-chat` from pairing to `chat.models` is a good idea to resolve this :thinking:

Reply:
I just had a look and the circular dependency comes from handle-sync-installation which call `start-public-chat`, wouldn't this create an endless loop of syncing messages between two devices?

In any case I would add pairing to handle-public-chat as well with a merge fx there.

Reply:
> wouldn't this create an endless loop of syncing messages between two devices?

I thought not if we set :sync? to false in `handle-public-chat`

Reply:
> In any case I would add pairing to handle-public-chat as well with a merge fx there.

not sure I get what you mean by that 😕

Reply:
but:

how about if, in here
https://github.com/status-im/status-react/blob/285df85d920bd3fd1fa76ef5193fd2c02e53bf57/src/status_im/pairing/core.cljs#L293

we dispatch `:chat.ui/start-public-chat` instead of calling `models.chat/start-public-chat`?

because that is the only line, as you have noticed, that is requiring `chat.models` in pairing namespace

Reply:
@bitsikka to solve #7992, instead of sync in start-public-chat with sync? parameter you move sync outside as it was before and add it for universal link using fx/merge

The long-term fix will probably involve separating sending pairing messages and receiving them

Reply:
I think I see. Ok, interpreting that in update PR pronto. Will ping ya once done